app/oauth.go

```diff
@@ -765,17 +765,27 @@ func (a *App) GetAuthorizationCode(w http.ResponseWriter, r *http.Request, servi
 
 	redirectURI := siteURL + "/signup/" + service + "/complete"
 
-	authURL := endpoint + "?response_type=code&client_id=" + clientId + "&redirect_uri=" + url.QueryEscape(redirectURI) + "&state=" + url.QueryEscape(state)
+	authURL, sErr := url.Parse(endpoint)
+	if sErr != nil {
+		return "", model.NewAppError("GetAuthorizationCode.ParseEndpoint", "api.user.get_authorization_code.endpoint_parse.app_error", nil, "failed to parse authentication endpoint URL", http.StatusBadRequest).Wrap(sErr)
+	}
+	queryParams := authURL.Query()
+	queryParams.Set("response_type", "code")
+	queryParams.Set("client_id", clientId)
+	queryParams.Set("redirect_uri", url.QueryEscape(redirectURI))
+	queryParams.Set("state", url.QueryEscape(state))
 
 	if scope != "" {
-		authURL += "&scope=" + utils.URLEncode(scope)
+		queryParams.Set("scope", utils.URLEncode(scope))
```

Comment:
In this case, the util will encode the space as `%20` instead of the default `+`, but it will also encode it twice. Since it's not a URL, it's probably not an issue in this case.